gui/executionGraphGui/client/apps/eg/src/app/components/file-browser/file-browser.component.html

```diff
@@ -3,6 +3,19 @@
 </ng-container>
 
 <ng-template #fileBrowserContainer>
+  <ng-container *ngIf="mode === 'save'">
+    <form class="filename-form">
+      <mat-form-field class="filename-input">
+        <input #fileNameInput matInput placeholder="File name" value="MyGraph.eg" autocomplete="off"/>
+      </mat-form-field>
+      <div class="save-button">
+        <button mat-button [disabled]="!isFileNameCorrect(fileNameInput.value)" (click)="saveFileName(fileNameInput.value)">
```

Comment:
This works. But is not actually the recommended way of doing Forms. https://angular.io/guide/reactive-forms elaborates on forms with validation. I'll try to add an example later...

Reply:
OK, thanks =)